### PR TITLE
Extract typed layout structures and strengthen layout tests (Fixes #25)

### LINKED ISSUE #25: Extract layout system and remove magic numbers from app shell
## Summary
Layout constants and coordinate calculations are embedded in the app shell and rely on magic values. This increases fragility and makes layout changes risky.

## Evidence
- Layout constants in src/main.rs lines 46-51.
- PTY layout computation in src/main.rs lines 69-96.

## Proposed changes
1. Move layout constants and calculations into a dedicated layout module.
2. Replace raw values with named typed structures (for example AppLayoutSpec).
3. Add property-style tests for boundary sizes and rounding behavior.
4. Add snapshot tests for representative terminal sizes.
5. Document layout invariants near the layout module.

## Acceptance criteria
- No direct layout magic numbers in app shell.
- Layout math has deterministic test coverage.
- Changes to layout can be made in one place with confidence.

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Closes #25.

The layout extraction (proposed change #1) was already in place via `src/layout.rs`; this PR completes the remaining proposed changes (#2–#5) from the issue.

### What changed

1. **Typed return structure (proposed change #2).** `compute_pty_layout` / `compute_pty_layout_inner` now return a named `PtyLayout` struct (`pty_rows`, `pty_cols`, `pane_col0`, `pane_row0`) instead of an ambiguous `(u16, u16, u16, u16)` tuple. This removes primitive obsession at the app-shell call sites (`src/main.rs`, `src/app_shell.rs`).

2. **AppLayoutSpec (proposed change #2).** A new `AppLayoutSpec` struct groups the fixed geometry constants into one typed contract. `AppLayoutSpec::DEFAULT` references the existing module constants so they remain the single source of truth (locked by a test).

3. **Property-style tests (proposed change #3).** Deterministic sweeps over boundary and representative terminal sizes assert the layout invariants (viewport dimensions >= 2, positive pane origins, half-up agent-row rounding, split partitioning). No external test crates added — uses the project's existing parametric-loop idiom.

4. **Golden/snapshot tests (proposed change #4).** A `GOLDEN_CASES` table pins the full `PtyLayout` for representative sizes (80x24, 120x40, 200x50, 60x20, 100x30, plus tight 10x10 / 20x8) in both fullscreen and windowed modes.

5. **Documented invariants (proposed change #5).** A module-level doc block enumerates the layout guarantees.

### Behavior

Numeric layout behavior is **unchanged** — this is a purely typing/test/docs refactor. All previously-passing tests still pass with identical expected values (adapted to struct field access).

### Acceptance criteria

- [x] No direct layout magic numbers in app shell.
- [x] Layout math has deterministic test coverage (property-style + golden tests).
- [x] Changes to layout can be made in one place with confidence (`AppLayoutSpec` + module constants).

### Verification

`make ci-check` passes locally:
- `cargo fmt --all --check`
- `scripts/check-clippy-allows.sh`
- `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- `cargo build --workspace --all-features --locked`
- `cargo test --workspace --all-features --locked` (all green)

### CODE REVIEW

File: src/layout.rs [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+//! # Layout invariants
+//!
+//! The geometry helpers below guarantee the following invariants for every
+//! input terminal size (including 0×0):
+//!
+//! - **Viewport never collapses**: `pty_rows` is always `>= 2` and `pty_cols`
+//!   is always `>= 2`. Degenerate inputs are clamped so the PTY viewport stays
+//!   usable.
+//! - **Pane origin is strictly positive**: `pane_col0` is always
+//!   `LEFT_COL_WIDTH + 1` (i.e. `> 0`) and `pane_row0` is always `> 0`.
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/layout.rs:14]: Good catch — fixed in 868d4c1. The pane origins are indeed zero-based render-grid coordinates (callers compare and subtract them directly against zero-based iocraft mouse coordinates, and the field names use the 0 suffix). Updated the module invariant block and the PtyLayout field docs accordingly.
